x/trigger/keeper/query_server.go

```diff
@@ -39,28 +43,23 @@ func (k Keeper) Triggers(ctx context.Context, req *types.QueryTriggersRequest) (
 
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
-	response := types.QueryTriggersResponse{}
-	kvStore := sdkCtx.KVStore(k.storeKey)
-	prefixStore := prefix.NewStore(kvStore, types.TriggerKeyPrefix)
-	pageResponse, err := query.FilteredPaginate(prefixStore, pagination, func(_ []byte, value []byte, accumulate bool) (bool, error) {
-		var trigger types.Trigger
-		vErr := trigger.Unmarshal(value)
-
-		if vErr != nil {
-			return false, vErr
-		}
-
-		if accumulate {
-			response.Triggers = append(response.Triggers, trigger)
-		}
-
-		return true, nil
-	})
-
+	triggers, pageRes, err := query.CollectionPaginate[uint64, types.Trigger, *collections.Map[uint64, types.Trigger], types.Trigger](
+		sdkCtx,
+		&k.TriggersMap,
+		pagination,
+		func(key uint64, trigger types.Trigger) (types.Trigger, error) {
+			trigger.Id = key
+			return trigger, nil
+		},
+	)
```

Comment:
Since all of the types are defined in the arguments provided to `CollectionPaginate`, you don't need to list them in the call.
```suggestion
	triggers, pageRes, err := query.CollectionPaginate(sdkCtx, &k.TriggersMap, pagination,
		func(key uint64, trigger types.Trigger) (types.Trigger, error) {
			trigger.Id = key
			return trigger, nil
		},
	)
```